src/main/java/duke/commands/Command.java

```diff
@@ -0,0 +1,162 @@
+package duke.commands;
+
+import duke.tasks.*;
+import duke.storage.Storage;
+import duke.ui.Ui;
+import duke.exception.DukeException;
+
+public class Command {
+    public enum Commands {
+        TASK("task"), TODO("todo"), EVENT("event"), DEADLINE("deadline"), LIST("list"), BYE("bye");
+
+        private String command;
+
+        Commands(String command) {
+            this.command = command;
+        }
+
+        public String getCommand() {
+            return this.command;
+        }
+    }
+
+    private String[] elements;
+    private String commandWord;
+    private String fullInput;
+    private boolean isExit = false;
+
+    public Command(String fullInput) {
+        this.elements = fullInput.split(" ");
+        this.commandWord = elements[0];
+        this.fullInput = fullInput;
+    }
+
+    public boolean isExit() {
+        return this.isExit;
+    }
+
+    public void execute(TaskList taskList, Storage storage, Ui ui) throws DukeException {
+        switch (this.commandWord) {
```

Comment:
Good use of the switch statement, instead of having if else blocks and using `equals()`
If you like to refactor this Command class, you could break up the commands into subclasses (e.g. DeleteCommand, ListCommand etc.)

Reply:
Yes, I think you could place `Commands` enum in a separate Java file, for better code readability.